packages/core/src/install/createPeerDependencyPatcher.ts

```diff
@@ -1,20 +1,27 @@
 import { PeerDependencyRules, ReadPackageHook } from '@pnpm/types'
+import matcher from '@pnpm/matcher'
 import isEmpty from 'ramda/src/isEmpty'
 
 export default function (
   peerDependencyRules: PeerDependencyRules
 ): ReadPackageHook {
-  const ignoreMissing = new Set(peerDependencyRules.ignoreMissing ?? [])
+  const ignoreMissingPatterns = [...new Set(peerDependencyRules.ignoreMissing ?? [])]
+  const ignoreVersionPatterns = [...new Set(peerDependencyRules.ignoreVersion ?? [])]
```

Comment:
I think `ignoreVersion` is just syntax sugar for allowedVersions. So, you can extend allowedVersions with the packages from `ignoreVersion` and make no changes in this function related to ignoreVersion